app/main.py

```diff
@@ -2,0 +1,45 @@
+class Animal:
+    def __init__(self, name: str,
+                 appetite: int = 0, is_hungry: bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        eaten = 0
+        if self.is_hungry:
+            eaten = self.appetite
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+        return eaten
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name)
+        self.appetite = 3
+        self.is_hungry = is_hungry
+
+    def catch_mouse(self):
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name)
+        self.name = name
```

Comment:
The assignment `self.name = name` is redundant here because it is already handled by the `super().__init__` method in the `Animal` class.